tests/test_security.py

```diff
@@ -0,0 +1,206 @@
+from html import escape
+from typing import TYPE_CHECKING
+
+import pytest
+
+from tbxforms.layout import Button
+
+from .forms import ButtonForm as ButtonFormBase
+from .forms import (
+    CheckboxesForm,
+    CheckboxForm,
+    DateInputForm,
+    FieldsetForm,
+    FileUploadForm,
+    RadiosForm,
+    SelectForm,
+    TextareaForm,
+    TextInputForm,
+)
+from .utils import render_form
+
+if TYPE_CHECKING:
+    # As we support Python 3.8 we have to use `typing.*` for tuples, dicts, etc
+    from typing import (
+        Callable,
+        Dict,
+        Tuple,
+        Type,
+    )
+
+    from django.forms import Form
+
+
+@pytest.mark.parametrize(
+    ("button_factory", "button_factory_args"),
+    (
+        (Button.primary, ("<i>name</i>", "<pre>Title</pre>")),
+        (Button.secondary, ("<i>name</i>", "<pre>Title</pre>")),
+        (Button.warning, ("<i>name</i>", "<pre>Title</pre>")),
+    ),
+)
+def test_html_escaping_buttons(
+    button_factory: "Callable",
+    button_factory_args: "Tuple[str, ...]",
+):
+    class ButtonForm(ButtonFormBase):
+        button_spec = (button_factory, button_factory_args)
+
+    form = ButtonForm()
+    rendered_form = render_form(form)
+
+    for button_arg in button_factory_args:
+        assert button_arg not in rendered_form
+        button_arg_escaped = escape(button_arg)
+        assert button_arg_escaped in rendered_form
+        # Let's also make sure we were testing the right value:
+        assert len(button_arg_escaped) > len(button_arg)
+
+
+@pytest.mark.parametrize(
```

Comment:
@kbayliss was it something like that you were thinking of? :slightly_smiling_face: